src/Servant/Auth/Hmac/Client.hs

```diff
@@ -122,8 +124,14 @@ servantRequestToPayload url sreq = RequestPayload
              fromList $ sort $ toList $ Servant.requestQueryString sreq
         }
 
-    host :: ByteString
-    host = Client.host req
+    hostAndPort :: ByteString
+    hostAndPort = case lookup (mk "Host") (Client.requestHeaders req) of
+        Just hp -> hp
+        Nothing ->
+            case (Client.secure req, Client.port req) of
+                (True, 443) -> Client.host req
+                (False, 80) -> Client.host req
```

Comment:
I don't know this codebase much but are you sure this will be reliable even when server is behind the reverse proxy that accepts https trafic but forwards unecrypted http trafic to haskell server?

Reply:
I guess this won't be a problem since this code is in client to server but I'm rather asking.

Reply:
I don't think unencrypted traffic will be an issue. With HMAC, we're more concerned about authentication and integrity rather than confidentiality. For the later, https should be used anyway. If https is not used, I guess confidentiality is not that important 😀